### PR TITLE
Expand shell variables, e.g. $HOME, in control file string entries

### PR DESCRIPTION
**Pull request by dwfncar**
Added melodies_monet/util/env_sub.
Can substitute env variables values, e.g. $HOME, $DATA_DIR, etc. in file strings.
New env_sub is used after reading any file path from the YAML file including
output_dir, model files, and obs filename.
If env var is not defined (getenv returns None), an error is printed and program exits.

### CODE REVIEW

**Comment by zmoon** [comment]
the new file is missing the license header

File: melodies_monet/util/env_sub.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    -------
+    file_str (str): modified file string
+    """
+    for subpath in file_str.split("/"):
+        if "$" in subpath:
+            envvar = subpath.replace("$", "")
+            envval = os.getenv(envvar)
+            if envval is None:
+                print("environment variable not defined: " + subpath)
+                exit(1)
```
Comment (zmoon) [inline melodies_monet/util/env_sub.py:22]: I think this should raise an Exception instead.
```suggestion
                raise Exception(f"Environment variable {envvar!r} detected in {file_str!r} not defined.")
```

File: melodies_monet/util/env_sub.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import os
+
+def env_sub(file_str):
+    """Substitute envirnoment variables prefixed by $
+    in file location string.
+
+    Parameters
+    -------
+    -------
```
Comment (zmoon) [inline melodies_monet/util/env_sub.py:9]:
```suggestion
    Parameters
    ----------
```

File: melodies_monet/util/env_sub.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    file_str (str): file string
```
Comment (zmoon) [inline melodies_monet/util/env_sub.py:10]:
```suggestion
    file_str : str
        File path, possibly containing ``$`` environment variables.
```
numpydoc format

File: melodies_monet/util/env_sub.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+    Returns
+    -------
+    file_str (str): modified file string
```
Comment (zmoon) [inline melodies_monet/util/env_sub.py:14]:
```suggestion
    file_str : str
        Modified file path.
```

File: melodies_monet/util/env_sub.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import os
+
+def env_sub(file_str):
+    """Substitute envirnoment variables prefixed by $
```
Comment (zmoon) [inline melodies_monet/util/env_sub.py:4]:
```suggestion
    """Substitute environment variables prefixed by ``$``
```

File: melodies_monet/util/env_sub.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    Parameters
+    -------
+    -------
+    file_str (str): file string
+
+    Returns
+    -------
+    file_str (str): modified file string
+    """
+    for subpath in file_str.split("/"):
```
Comment (zmoon) [inline melodies_monet/util/env_sub.py:16]: To be safe / for OS compat. https://stackoverflow.com/q/3167154
```suggestion
    from pathlib import Path

    for subpath in Path(file_str).parts:
```

**Review by zmoon — COMMENTED** [review]
Note also stdlib [`os.path.expandvars()`](https://docs.python.org/3/library/os.path.html#os.path.expandvars), in case you didn't already consider it.

**Comment by dwfncar** [comment]
os.path.expandvars works

**Comment by zmoon** [comment]
I suspect @rschwant would want you to add a note in the docs.

**Comment by dwfncar** [comment]
Calling expandvars consistently now, wrapping extraction from self.control_dict in all cases.

File: docs/appendix/yaml.rst [hunk 7]
```diff
@@ -39,7 +39,8 @@ All input related to the analysis class.
 (e.g., '2019-08-03-12:00:00')
 
 **output_dir**: This is optional. This is the directory where the plots are saved. 
-If not specified plots stored in code directory. 
+If not specified plots stored in code directory.
+Shell variables prefixed with the $ symbol, such as $HOME, will be expanded.
```
Comment (zmoon) [inline docs/appendix/yaml.rst:43]:
```suggestion
Shell variables prefixed with the ``$`` symbol, such as ``$HOME``, will be expanded.
```

Comment (zmoon) [reply docs/appendix/yaml.rst:43]: And for the other instances as well...

**Comment by dwfncar** [comment]
git commit expanded $HOME to /Users/davidfillmore in my last commit message

**Review by zmoon — APPROVED** [review]

**Review by rschwant — APPROVED** [review]
Looks good!

**Comment by zmoon** [comment]
CI failure after merge is explained in #97

**Comment by zmoon** [comment]
@dwfncar feel free to delete the branch if you are done with it

**Pull request merged**